### PR TITLE
Remove all deprecationNotices components/actions/selectors/reducers.

### LINKED ISSUE #851: Increase spacing in notifications
The server-side notifications could do with a little bit of breathing room between the message, the expander icon and the dismiss link.

image

### LINKED ISSUE #1083: Duplicated deprecation notifications
Deprecations are now coming through via the `notifications` api *and* the `deprecations` api. We should remove the deprecations actions/reducers/selectors and replace with notifications (and fix the styling).

Screenshot_2020-05-04 Machines bolla MAAS(1)

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done
- Removed all deprecationNotices components/actions/selectors/reducers as deprecations now just use the standard notifications api.
- Improved NotificationGroup styling.
- Added dismissable/non-dismissable functionality. If a notification is not dismissable the "Dismiss" link is not shown. If it's part of a group, the "Dismiss all" link will still show but it won't run for that notification.

## QA
- You will need to set up a 2.8 MAAS that was initialised in `all` mode, i.e. the database has been set up inside the snap. This mode has been deprecated in 2.8, so you need to set up a 2.7 MAAS first.
  - Install 2.7 MAAS: `sudo snap install maas --channel=2.7/stable`
  - Initialise in all mode: `sudo maas init` then choose mode `all` and set up a first user
  - Update to 2.8: `sudo snap refresh maas --channel=2.8/edge`
- Checkout maas-ui on master and point the UI at the new MAAS (change the MAAS_URL in `/proxy/env.local`).
- Go through the intro steps until you get to the machine list.
- confirm that there are two deprecation notices, one dismissable and one not.
- Checkout this branch.
- Check that there is only one non-dismissable warning notification about the db being set up inside the snap. You'll probably have another warning about DHCP too, so you will need to expand the warning list.
- If you do have more than one warning, click "Dismiss all" and check that the db deprecation notification remains.
- Run the db migration script: `sudo bash -c '/snap/maas/current/helpers/migrate-snap-database'`
- Wait a minute for MAAS to restart and refresh the machine list. If it doesn't restart automatically, run `sudo snap restart maas`
- Check that the db deprecation warning no longer shows 

## Fixes
Fixes #851 
Fixes #1083 

### Before
bolla internal_5240_MAAS_r_machines (1)

### After
0 0 0 0_8400_MAAS_r_machines (1)

### CODE REVIEW

**Comment by Caleb-Ellis** [comment]
@albertodonato For a while deprecations were coming through via the notifications websocket api, but I can't seem to get it back. Any ideas why?

**Comment by albertodonato** [comment]
@Caleb-Ellis are you talking about all notifications, or specifically about the deprecation one? If you're testing for the deprecation one on bolla, I believe @sparkiegeek has migrated it to have an external database, so the deprecation is not there anymore.

**Comment by Caleb-Ellis** [comment]
@albertodonato ahh right, I didn't realise the db had been migrated out of the snap on bolla! Thanks.

**Comment by Caleb-Ellis** [comment]
Don't review yet - I'll add the dismissable stuff in as part of this PR.

File: legacy/src/app/directives/notifications.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ export function maasNotifications(NotificationsManager, ManagerHelperService) {
 
       scope.dismissCategory = category => {
         const categoryNotifications = scope.notifications.filter(
-          notification => notification.category === category
+          (notification) =>
+            notification.category === category && notification.dismissable
```
Comment (squidsoup) [inline legacy/src/app/directives/notifications.js:45]: Nice to see these changes to the notification API, thanks @albertodonato :)

File: ui/src/app/base/components/NotificationGroupMessage/NotificationGroupMessage.js [hunk 2]
```diff
@@ -6,23 +6,28 @@ import { Button } from "@canonical/react-components";
...
-          {action}
-        </Button>
-      )}
-    </p>
+    <div className="p-notification__response u-no-max-width">
+      <ul className="p-inline-list u-no-margin--bottom">
+        <li className="p-inline-list__item">
+          <span
+            data-test="notification-message"
+            dangerouslySetInnerHTML={{ __html: message }}
```
Comment (squidsoup) [inline ui/src/app/base/components/NotificationGroupMessage/NotificationGroupMessage.js:14]: Maybe in future the API could return `message` and `url` so we don't have to do this.

**Pull request merged**